test/ILLink.RoslynAnalyzer.Tests/TestCaseUtils.cs

```diff
@@ -68,11 +78,27 @@ static bool IsWellKnown (AttributeSyntax attr)
 			}
 		}
 
-		public static void RunTest<TAnalyzer> (MemberDeclarationSyntax m, List<AttributeSyntax> attrs, params (string, string)[] MSBuildProperties)
+		public static async Task<Compilation> GetCompilation (string src)
+		{
+			var srctree = CSharpSyntaxTree.ParseText (src);
+			var mdRef = MetadataReference.CreateFromFile (typeof (Mono.Linker.Tests.Cases.Expectations.Metadata.BaseMetadataAttribute).Assembly.Location);
+			var comp = CSharpCompilation.Create (
```

Comment:
This could probably be a good place to call a method that pulls in dependencies. Probably by understanding the `SetupCompileBefore`/`SetupCompileAfter` attributes.